plugins/mac-capture/mac-sck-audio-capture.m

```diff
@@ -0,0 +1,386 @@
+		os_sem_wait(sc->shareable_content_available);
+		[sc->shareable_content release];
+		os_sem_destroy(sc->shareable_content_available);
+		sc->shareable_content_available = NULL;
+	}
+
+	if (sc->capture_delegate) {
+		[sc->capture_delegate release];
+	}
+	[sc->application_id release];
+
+	pthread_mutex_destroy(&sc->mutex);
+	bfree(sc);
+}
+
+static bool init_audio_screen_stream(struct screen_capture *sc)
+{
+	SCContentFilter *content_filter;
+
+	sc->stream_properties = [[SCStreamConfiguration alloc] init];
+	os_sem_wait(sc->shareable_content_available);
+
+	SCDisplay * (^get_target_display)() = ^SCDisplay *()
+	{
+		__block SCDisplay *target_display = nil;
+		[sc->shareable_content.displays
+			indexOfObjectPassingTest:^BOOL(
+				SCDisplay *_Nonnull display, NSUInteger idx,
+				BOOL *_Nonnull stop) {
+				if (display.displayID == sc->display) {
+					target_display = sc->shareable_content
+								 .displays[idx];
+					*stop = TRUE;
+				}
+				return *stop;
+			}];
+		return target_display;
+	};
+
+	void (^set_display_mode)(struct screen_capture *, SCDisplay *) = ^void(
+		struct screen_capture *sc2, SCDisplay *target_display) {
+		CGDisplayModeRef display_mode =
+			CGDisplayCopyDisplayMode(target_display.displayID);
+		[sc2->stream_properties
+			setWidth:CGDisplayModeGetPixelWidth(display_mode)];
+		[sc2->stream_properties
+			setHeight:CGDisplayModeGetPixelHeight(display_mode)];
+		CGDisplayModeRelease(display_mode);
+	};
+
+	switch (sc->capture_type) {
+	case ScreenCaptureAudioDesktopStream: {
+		SCDisplay *target_display = get_target_display();
+
+		content_filter = [[SCContentFilter alloc]
+			 initWithDisplay:target_display
+			excludingWindows:[[NSArray alloc] init]];
+
+		set_display_mode(sc, target_display);
+	} break;
+	case ScreenCaptureAudioApplicationStream: {
+		SCDisplay *target_display = get_target_display();
+		__block SCRunningApplication *target_application = nil;
+		{
+			[sc->shareable_content.applications
+				indexOfObjectPassingTest:^BOOL(
+					SCRunningApplication
+						*_Nonnull application,
+					NSUInteger idx, BOOL *_Nonnull stop) {
+					if ([application.bundleIdentifier
+						    isEqualToString:
+							    sc->
+							    application_id]) {
+						target_application =
+							sc->shareable_content
+								.applications
+									[idx];
+						*stop = TRUE;
+					}
+					return *stop;
+				}];
+		}
+		NSArray *target_application_array = [[NSArray alloc]
+			initWithObjects:target_application, nil];
+
+		content_filter = [[SCContentFilter alloc]
+			      initWithDisplay:target_display
+			includingApplications:target_application_array
+			     exceptingWindows:[[NSArray alloc] init]];
+
+		set_display_mode(sc, target_display);
+	} break;
+	}
+	os_sem_post(sc->shareable_content_available);
+	[sc->stream_properties setQueueDepth:8];
+
+	[sc->stream_properties setCapturesAudio:TRUE];
+	[sc->stream_properties setExcludesCurrentProcessAudio:TRUE];
+	[sc->stream_properties setChannelCount:2];
```

Comment:
If we're refactoring SCK around this, maybe investigate if this is bound to always be stereo, or if we could pass the channel count configured in OBS here.

Reply:
According to the [docs](https://developer.apple.com/documentation/screencapturekit/scstreamconfiguration/3931901-channelcount), acceptable values are just mono or stereo. Looks like I could at least handle the mono case as well.

Reply:
Force pushed so this became outdated, my git usage is a bit hamfisted, sorry if I did something wrong. I replaced the last line with the [following block](https://github.com/obsproject/obs-studio/blob/b4c920bda7174292fc1d0d7867e87c1c57fadf4e/plugins/mac-capture/mac-sck-audio-capture.m#L137-L150):

```objc
        [sc->stream_properties setCapturesAudio:TRUE];
	[sc->stream_properties setExcludesCurrentProcessAudio:TRUE];

	struct obs_audio_info audio_info;
	BOOL did_get_audio_info = obs_get_audio_info(&audio_info);
	if (!did_get_audio_info) {
		MACCAP_ERR(
			"init_audio_screen_stream: No audio configured, returning %d\n",
			did_get_audio_info);
		return did_get_audio_info;
	}
	int channel_count = get_audio_channels(audio_info.speakers);
	if (channel_count == 1) {
		[sc->stream_properties setChannelCount:1];
	} else {
		[sc->stream_properties setChannelCount:2];
	}
```

If this looks acceptable I'll let you resolve.

Reply:
Logically this has the potential to set the `channelCount` to `2`, even though the actual count is `0`, which would be logically wrong.

A simpler solution could be to use something along the lines of `[sc->stream_properties setChannelCount:((channel_count > 1) ? 2 : channel_count)];`.

Reply:
Resolved with:

https://github.com/obsproject/obs-studio/blob/839a86c40e7c7085337ed0268a5a00f3d764a362/plugins/mac-capture/mac-sck-audio-capture.m#L148-L152